src/lib/core/Global.h

```diff
@@ -97,25 +97,70 @@ class Global
     ~Global()          = default;
 
 private:
+#if CHIP_CONFIG_DYNAMICALLY_ALLOCATED_GLOBALS
+    std::remove_extent_t<T> * mStorage = nullptr;
```

Comment:
There don't seem to be any strong use cases for `Global` with arrays, so I would prefer to just `static_assert` here that T isn't an array and have the code be simpler.